wolfssl/internal.h

```diff
@@ -2784,7 +2799,12 @@ WOLFSSL_LOCAL int TLSX_AddEmptyRenegotiationInfo(TLSX** extensions, void* heap);
  */
 typedef struct TicketNonce {
     byte len;
-    byte data[MAX_TICKET_NONCE_SZ];
+#ifdef WOLFSSL_TICKET_NONCE_MALLOC
+    byte *data;
```

Comment:
Maybe the ticket nonce should be kept in the `WOLFSSL_SESSION` object?

Reply:
> Any member larger than 1 byte will cause alignment errors. This is because the data read from the network layer is directly cast to `ExternalTicket` in `DoClientTicket()`. The comments above `InternalTicket` and `ExternalTicket` specify this.
> 
> See [`617eda9` (#4807)](https://github.com/wolfSSL/wolfssl/pull/4807/commits/617eda9d449a3755e9791ae8a8ce26ff43a7304b) for context.

Should the padding of the external ticket assures that it will be aligned?

Reply:
> Also having `data` be either a pointer or an array can be dangerous. `sizeof` will return different values for different configs. I think this can also facilitate making mistakes.

I agree with you, I thought to save the space of a `lenAlloc` variable. I can refactor everything now, but it will take some time.

Reply:
> Maybe the ticket nonce should be kept in the `WOLFSSL_SESSION` object?

Do you mean, without the struct? what really changes?

Reply:
> Should the padding of the external ticket assures that it will be aligned?

No it used for the crypto meta data (padding). When we cast the input into a `ExternalTicket` then `TicketNonce.data` can be misaligned and cause exceptions.